spring-cloud-gcp-pubsub-stream-binder/src/main/java/com/google/cloud/spring/stream/binder/pubsub/provisioning/PubSubChannelProvisioner.java

```diff
@@ -71,32 +74,35 @@ public ConsumerDestination provisionConsumerDestination(String topicName, String
 			ExtendedConsumerProperties<PubSubConsumerProperties> properties)
 			throws ProvisioningException {
 
-		Topic topic = makeSureTopicExists(topicName, properties.getExtension().isAutoCreateResources());
+		// topicName may be either the short or fully-qualified version.
+		TopicName topicNameObj = PubSubTopicUtils.toTopicName(topicName, this.pubSubAdmin.getProjectId());
```

Comment:
If you wanted to avoid introducing the extra `getProjectId` on pubsubadmin, you could just compute the `topicShortName` and then continue passing `topicName` to `this.pubSubAdmin.createSubscription(..)` since it appears to accept both short and fully qualified names (so no need for `topicFullyQualifiedName` and getting the project Id).

```
String topicShortName = TopicName.isParsableFrom(topicName) ? TopicName.parse(topic).getTopic() : topicName;
```

Reply:
+1, we just need to extract the short name.

Reply:
Nice - I had to rework a few other things to get this to work, but it's a bit nicer now.